torch/_guards.py

```diff
@@ -520,6 +525,20 @@ def current_frame(frame_summary):
         finally:
             tc.frame_summary_stack.pop()
 
+    @staticmethod
+    @contextlib.contextmanager
+    def report_output_strides():
+        tc = TracingContext.get()
+        if tc is None:
+            yield None
+            return
+        old_output_strides = tc.output_strides
```

Comment:
Just want to raise a concern about nested compiling where we may need a stack for output_strides. But I can not come up with a realistic example. So probably the current implementation is good enough

Reply:
This implicitly is a stack via old_output_strides!

Reply:
ah, right, that's the whole point of context manager after all... haha